contracts/src/AxelarGateway.sol

```diff
@@ -168,67 +161,106 @@ abstract contract AxelarGateway is IAxelarGateway, EternalStorage {
     ) internal {
         uint256 mintLimit = tokenDailyMintLimits(symbol);
         uint256 mintAmount = tokenDailyMintAmounts(symbol);
-        require((mintLimit == 0) || (mintLimit >= mintAmount + amount), 'EXCEED_LIMIT');
+        require((mintLimit == uint256(0)) || (mintLimit >= mintAmount + amount), 'EXCEED_LIMIT');
 
-        address tokenAddr = tokenAddresses(symbol);
-        require(tokenAddr != address(0), 'TOKEN_NOT_EXIST');
+        address tokenAddress = tokenAddresses(symbol);
+        require(tokenAddress != address(0), 'TOKEN_NOT_EXIST');
 
-        BurnableMintableCappedERC20(tokenAddr).mint(account, amount);
+        BurnableMintableCappedERC20(tokenAddress).mint(account, amount);
         _setTokenDailyMintAmount(symbol, mintAmount + amount);
     }
 
     function _burnToken(string memory symbol, bytes32 salt) internal {
-        address tokenAddr = tokenAddresses(symbol);
-        require(tokenAddr != address(0), 'TOKEN_NOT_EXIST');
+        address tokenAddress = tokenAddresses(symbol);
+        require(tokenAddress != address(0), 'TOKEN_NOT_EXIST');
 
-        new Burner{ salt: salt }(tokenAddr, salt);
+        new Burner{ salt: salt }(tokenAddress, salt);
     }
 
-    function _update(address newVersion, bytes memory setupParams) internal {
-        bytes memory proposedUpdate = getBytes(KEY_PROPOSED_UPDATE);
-        require(proposedUpdate.length != 0, 'NO_PPS');
+    function _update(address newImplementation, bytes memory setupParams) internal {
+        bytes memory _proposedUpdate = proposedUpdate();
+        require(_proposedUpdate.length != 0, 'NO_PPS');
 
-        deleteBytes(KEY_PROPOSED_UPDATE);
-        deleteUint(KEY_PROPOSED_UPDATE_TIME);
+        _clearProposedUpdate();
+        _setProposedUpdateTime(uint256(0));
 
-        if (keccak256(proposedUpdate) != keccak256(abi.encodePacked(newVersion, setupParams))) return;
+        // NOTE: Any attempt to “incorrectly” update is the way to clear the pending update, since we return rather than revert.
+        if (keccak256(_proposedUpdate) != keccak256(abi.encodePacked(newImplementation, setupParams))) return;
 
-        (bool success, ) = newVersion.delegatecall(abi.encodeWithSelector(IAxelarGateway.setup.selector, setupParams));
+        // AUDIT: If `newImplementation.setup` performs `selfdestruct`, it will result in the loss of _this_ implementation (thereby losing the gateway)
+        //        if `_update` is entered within the context of _this_ implementation itself. Consider directly calling `forceUpdate`.
+        (bool success, ) =
+            newImplementation.delegatecall(abi.encodeWithSelector(IAxelarGateway.setup.selector, setupParams));
         require(success, 'SETUP_FAILED');
 
-        setAddress(KEY_IMPLEMENTATION, newVersion);
+        _setImplementation(newImplementation);
 
-        emit Updated(address(this), newVersion);
+        emit Updated(address(this), newImplementation);
```

Comment:
Maybe let's pick up the `Upgraded` event defined in https://eips.ethereum.org/EIPS/eip-1967? `address(this)` here would be the proxy itself so it's not correct anyway.

Reply:
Only if you're ok with `proposeUpdate` being changed to `proposeUpgrade`. But I do agree on the standardized `Upgraded` event.

Reply:
yes I'll do that.